File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/mutate/dao/RecordMutateDtoDao.java
package ru.citeck.ecos.records3.record.op.mutate.dao;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import ru.citeck.ecos.records2.RecordRef;

public interface RecordMutateDtoDao<T> {

    @Nullable
    T getRecToMutate(@NotNull RecordRef recordRef);

    void saveMutatedRec(@NotNull T record);
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/mutate/dao/RecordMutateDtoDao.kt
package ru.citeck.ecos.records3.record.op.mutate.dao

import ru.citeck.ecos.records2.RecordRef

interface RecordMutateDtoDao<T> {

    fun getRecToMutate(record: RecordRef): T

    fun saveMutatedRec(record: T)
}
